src/repositories/gitlab.rs

```diff
@@ -0,0 +1,222 @@
+use crate::error::Result;
+use chrono::{DateTime, Utc};
+use log::warn;
+use reqwest::{
+    blocking::Client as HttpClient,
+    header::{HeaderMap, HeaderValue, ACCEPT, AUTHORIZATION, USER_AGENT},
+};
+use serde::Deserialize;
+use std::collections::HashSet;
+
+use crate::repositories::{
+    FetchRepositoriesResult, Repository, RepositoryForge, RepositoryName, APP_USER_AGENT,
+};
+
+const GRAPHQL_UPDATE: &str = "query($ids: [ID!]!) {
+    projects(ids: $ids) {
+        nodes {
+            id
+            nameWithNamespace
+            lastActivityAt
+            description
+            starCount
+            forksCount
+            openIssuesCount
+        }
+    }
+}";
+
+const GRAPHQL_SINGLE: &str = "query($fullPath: ID!) {
+    project(fullPath: $fullPath) {
+        id
+        nameWithNamespace
+        lastActivityAt
+        description
+        starCount
+        forksCount
+        openIssuesCount
+    }
+}";
+
+pub struct GitLab {
+    client: HttpClient,
+    host: &'static str,
+}
+
+impl GitLab {
+    pub fn new(host: &'static str, access_token: &Option<String>) -> Result<Option<Self>> {
```

Comment:
There is no need to return `Option<Self>` here, as GitLab also works without an API token.